src/services/ApiService.ts

```diff
@@ -185,58 +186,65 @@ export class ApiService {
         // Upgrade to V2 if required
-        if (preferences?.version && preferences.version === 1) {
-            if (preferences.astColor || preferences.astColor === 0) {
-                switch (preferences.astColor) {
-                    case 0:
-                        preferences.astColor = "auto-black";
-                        break;
-                    case 1:
-                        preferences.astColor = "auto-white";
-                        break;
-                    case 2:
-                        preferences.astColor = "auto-red";
-                        break;
-                    case 3:
-                        preferences.astColor = "auto-forest";
-                        break;
-                    case 4:
-                        preferences.astColor = "auto-blue";
-                        break;
-                    case 5:
-                        preferences.astColor = "auto-turquoise";
-                        break;
-                    case 6:
-                        preferences.astColor = "auto-violet";
-                        break;
-                    case 7:
-                        preferences.astColor = "auto-gold";
-                        break;
-                    case 8:
-                        preferences.astColor = "auto-gray";
-                        break;
-                    default:
-                        preferences.astColor = "auto-blue";
-                        break;
-                }
-            }
-            preferences.version = 2;
-            this.setPreferences(preferences);
+        if (preferences["version"] === 1) {
+            // Convert preferences[PreferenceKeys.WCS_OVERLAY_AST_COLOR] from a number in version 1 to a string in version 2
+            // default to "auto-blue" if the value is not in the AST_COLORS map
+            enum ASTColors {
+                black,
+                white,
+                red,
+                forest,
+                blue,
+                turquoise,
+                violet,
+                gold,
+                gray
+            }
+            const astColorKey = PreferenceKeys.WCS_OVERLAY_AST_COLOR;
+            const color = preferences[astColorKey] in ASTColors ? ASTColors[preferences[astColorKey]] : "blue";
+            preferences[astColorKey] = `auto-${color}`;
+            this.setPreference(astColorKey, preferences[astColorKey]);
+
+            // preferences["version"] = 2;
+            this.setPreference("version", 2);
+        }
+
+        // Upgrade to schema version 2
+        let key = PreferenceKeys.WCS_OVERLAY_WCS_TYPE;
+        if (preferences[key]) {
+            preferences[key] = preferences[key].toLowerCase();
+            this.setPreference(key, preferences[key]);
+        }
+
+        key = PreferenceKeys.IMAGE_PANEL_MODE;
+        if (preferences[key] === "static") {
+            preferences[key] = "fixed";
+            this.setPreference(key, preferences[key]);
```

Comment:
I think we agreed to leave this out (the bad pref will just be deleted in the validation step).